File: go-backend/models/fact.go
package models

import "time"

// Fact represents a single extracted fact stored for a user + card.
type Fact struct {
	ID        int       `json:"id"`
	UserID    int       `json:"user_id"`
	CardPK    int       `json:"card_pk"`
	Fact      string    `json:"fact"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: zettelkasten-front/src/models/Fact.ts
import { PartialCard } from "./Card";

export interface Fact {
  id: number;
  fact: string;
  created_at: string;
  updated_at: string;
}

export interface FactWithCard {
  id: number;
  fact: string;
  created_at: string;
  updated_at: string;
  card: PartialCard;
}
